### PR TITLE
Update solar.json with length

### PR DESCRIPTION
**Pull request by Chroniclerope**
#### Summary
Balance "Add proper length to solar panels and cells"

#### Purpose of change
Currently solar panels have no length given to them, allowing for survivors to fit these inside of backpacks.

#### Describe the solution
Given the weight of the solar panel item at 30 pounds, looking at real life counter parts this leads to an estimate of roughly 1 meter by .75 meter panel with the addition of a mount and fixtures taking up weight, therefore making the longest side 100 cm.  This was due to the crafting recipie taking 12 solar cells, making a 4x3 configuration the best possibility for a solar panel in game.

#### Describe alternatives you've considered
Overall the solar panels could be smaller given their low wattage, but then the weight should also be reduced, as a 30 pound panel is far too much for anything smaller than currently proposed

#### Testing
Tested the code on my personal branch, works fine.

### CODE REVIEW

File: data/json/items/vehicle/solar.json [hunk 1]
```diff
@@ -70,6 +71,7 @@
     "symbol": ";",
     "volume": "250 ml",
     "category": "spare_parts",
+    "longest_side": "25 cm",
```
Comment (wapcaplet) [inline data/json/items/vehicle/solar.json:74]: I understand you arrived at this value based on a 4x3 configuration of cells in the overall panel, but I don't think real-life cells are nearly this large. Something closer to 6x6 inches (~15x15 cm) is more typical of the cells in real-life panels. [Reference](https://unboundsolar.com/blog/solar-panel-size-guide)

Since the recipe only takes 12 of these, that would suggest the total panel size should be much smaller as well.

Comment (Chroniclerope) [reply data/json/items/vehicle/solar.json:74]: Alright, so supposing its only a 60x45cm solar panel, that would roughly put its weight around 7 pounds with a metal frame and converter.

Comment (wapcaplet) [reply data/json/items/vehicle/solar.json:74]: Yeah that sounds reasonable.

Another piece of info you may have already considered is that the standard solar panel vehicle part generates 50 watts of power. This would seem to align well with having 12 cells in the panel, based on [this source](https://letsavelectricity.com/what-is-the-size-of-a-solar-panel-weight-of-a-solar-panel/) which says "The standard size of a 250 watt 60 cell solar panel is 39 inch x 66 inch" (167x100 cm ~= 16,700 cm^2) - so dimensions around one-fifth that area should be about right for 50 watts (60x45cm ~= 2700 cm^2, pretty close), and it's exactly one-fifth the number of cells.

The individual solar cell weight of 500g seems a bit steep for 15x15cm, so that could be reduced if it makes the math come out a little better.

Comment (Chroniclerope) [reply data/json/items/vehicle/solar.json]: I've adjusted the values, reducing the panel itself to 60 cm and 3200 grams, while solar cells are down to 15 cm and 200 grams.  Given twelve cells are present, this gives a total weight of 2400g from the cells alone, leaving 800g to be taken up by the power converter and frame.

**Comment by wapcaplet** [comment]
I was a bit surprised to see this cause a test failure, but it turns out we have some assertions about vehicle weight and efficiency for electric cars in `tests/vehicle_efficiency_test.cpp`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/946ed1dbe70210a478118cde9771feaf015fb288/tests/vehicle_efficiency_test.cpp#L439

The failure from CI (Basic Build and Test GCC 9, Curses, LTO) looks like this:

```
(~[slow] ~[.])=> ../tests/vehicle_efficiency_test.cpp:197: FAILED:
(~[slow] ~[.])=>   CHECK( to_gram( veh.total_mass() ) == expected_mass )
(~[slow] ~[.])=> with expansion:
Error: (~[slow] ~[.])=>   905243 (0xdd01b) == 948443 (0xe78db)
```

The expected `electric_car` vehicle weight is off by 43,200 grams (948443-905243). This makes sense, because you reduced the weight of each panel by 10,800 (from 14,000 to 3200), so four of those would weigh 43,200 less. You should easily be able to fix this by replacing `948443` with `905243` wherever it appears in `tests/vehicle_efficiency_test.cpp`.

Another side-effect appears to have been the "adjusted_tiles_travelled" assertion, which are coming up higher than expected. Those failures look like this:

```
(~[slow] ~[.])=> ../tests/vehicle_efficiency_test.cpp:263: FAILED:
(~[slow] ~[.])=>   CHECK( adjusted_tiles_travelled <= max_dist )
(~[slow] ~[.])=> with expansion:
Error: (~[slow] ~[.])=>   26027 (0x65ab) <= 25270 (0x62b6)
```

In other words, you made the electric car somewhat more efficient by reducing its weight, so now this `<=` expectation is higher than before. I think these are related to the last two numbers passet to `test_vehicle`, being the expected distance (?) covered on both pavement and dirt terrain, with frequent stops and starts. I don't quite get all the details, but this part is doing a crude attempt at margin-of-error, and the expected limit `25270` is coming from the second-to-last argument to `test_vehicle` for the electric car, namely `25020`.

Long story short, to fix this test failure, adjust that `25020` closer to the first value in the error message - `26027` in this case. If it continues to fail (doubtful) adjust again until it behaves consistently.

**Comment by wapcaplet** [comment]
Try these values for the first section in `tests/vehicle_efficiency_test.cpp` (line 440):

```diff
-    test_vehicle( "electric_car", 948443, 372500, 231100, 25020, 14280 );
+    test_vehicle( "electric_car", 905243, 382840, 252690, 26030, 16190 );
```

And these for the `// in reverse` section, line 459 (only the first number needs to be changed):

```diff
-    test_vehicle( "electric_car", 948443, 373700, 231800, 25070, 14310, 0, 0, true );
+    test_vehicle( "electric_car", 905243, 373700, 231800, 25070, 14310, 0, 0, true );
```

**Comment by I-am-Erk** [comment]
Are you around to fix those tests, chroniclerope?

**Comment by Chroniclerope** [comment]
Oh, sorry.  I got caught up in some stuff in my life, I'll see about this in the morning.

File: data/json/items/vehicle/solar.json (hunk 1 above)
Comment (Alex-Folts) [reply data/json/items/vehicle/solar.json]: Usually solar panels covered with glass, 60x45cm 3mm thick sheet of glass would weight about 2kg. But solar panel recipe don't contain glass, maybe oversight. 

As an option we could add glass to the recipe and adjust total mass of the solar panel. So it wasn't too lightweight, afterall glass in there irl for a reason.

**Comment by Maleclypse** [comment]
I'm not expecting the tests to pass after merging conflicts but hopefully it will make it easier to see where the test is failing.

**Comment by oosyrag** [comment]
Some discussion on solar panels in discord:
https://discord.com/channels/598523535169945603/598529174302490644/945190870238498876

Summary - Have a solar panel appliance as a new item, that represents rooftop solar panels (2 tiles, ~3'x6'). It would roughly consist of 8 of the current solar panel item, or four per tile. Basically representing a more dense arrangement of solar cells in a tile. The current solar panel can remain as is on vehicles, due to vehicles tiles not really scaling well with the m^2 per tile dimensions. It would represent a less dense solar cell arrangement per tile, by a factor of 2, with the justification that our in game vehicles are very roughly twice as big as they should be per tile (4m-5m wide vehicle vs 2m wide vehicles IRL, going by m^2 per tile). 

If the current solar panel is 50W, that would result in 200W per tile appliance, or a 400W "full solar panel", which is on the high side, but can be tweaked. The weight discrepancies could also be resolved by having 2 types of solar panels, as rooftop panels can have additional frame or glass requirements to build. The number of cells per solar panel item could also be adjusted for example to 9 and the wattage reduced accordingly to fit better.

My further recommendation would be to eventually remove cars with solar panels, and have a rare chance to find a house with rooftop solar instead. RVs and boats would be the exception for having vehicle solar panels.

9 cells = 1 solar panel item = 40W = 1 vehicle tile solar panel
36 cells = 4 solar panel items = 160W = 1 tile rooftop solar panel appliance
72 cells = 8 solar panel items = 320W = 2 tile rooftop solar panel appliances = 1 IRL rooftop solar panel
